internal/services/dashboard/dashboard_grafana_managed_private_endpoint_resource.go

```diff
@@ -0,0 +1,277 @@
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Dashboard.ManagedPrivateEndpointsClient
+			id, err := managedprivateendpoints.ParseManagedPrivateEndpointID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			grafanaId, err := GetGrafanaIdFromManagedPrivateEndpointId(id.ID())
+			if err != nil {
+				return err
+			}
+
+			state := ManagedPrivateEndpointModel{
+				Name:      id.ManagedPrivateEndpointName,
+				GrafanaId: grafanaId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Tags = pointer.From(model.Tags)
+
+				if props := model.Properties; props != nil {
+					state.GroupIds = pointer.From(props.GroupIds)
+					state.PrivateLinkResourceId = pointer.From(props.PrivateLinkResourceId)
+					state.PrivateLinkResourceRegion = pointer.From(props.PrivateLinkResourceRegion)
+					state.RequestMessage = pointer.From(props.RequestMessage)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r ManagedPrivateEndpointResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Dashboard.ManagedPrivateEndpointsClient
+			id, err := managedprivateendpoints.ParseManagedPrivateEndpointID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ManagedPrivateEndpointResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Dashboard.ManagedPrivateEndpointsClient
+
+			id, err := managedprivateendpoints.ParseManagedPrivateEndpointID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ManagedPrivateEndpointModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", id)
+			}
+			properties := resp.Model
+
+			if metadata.ResourceData.HasChange("tags") {
+				properties.Tags = &model.Tags
+			}
+
+			if err := client.CreateThenPoll(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
```

Comment:
In nearly all cases the `Model` has a field called `Properties` which is why we follow the naming convention of `model := resp.Model` and `props := model.Properties`. For clarity and consistency can you please update this to 
```suggestion
			model := resp.Model

			if metadata.ResourceData.HasChange("tags") {
				model.Tags = &model.Tags
			}

			if err := client.CreateThenPoll(ctx, *id, *model); err != nil {
				return fmt.Errorf("updating %s: %+v", *id, err)
			}
```

Reply:
@stephybun I believe I have resolved this, but please double check; I had to modify the variable assigned to the model created in line 235. I hope this maintains consistency.